### ISSUE TITLE
Sift SDK is accessing UI methods from background thread

### ISSUE THREAD

**Issue by edenman** [SiftScience/sift-ios#61]
```
Main Thread Checker: UI API called on a background thread: -[UIApplication applicationState]
PID: 98610, TID: 1552400, Thread name: (none), Queue name: com.sift.SFUploader, QoS: 0
Backtrace:
4   Sift                                0x000000010851874c -[SFUploader doUpload] + 76
5   Sift                                0x0000000108517aec __58-[SFUploader initWithArchivePath:sift:config:backoffBase:]_block_invoke + 44
6   libdispatch.dylib                   0x000000010daf06c8 _dispatch_client_callout + 8
7   libdispatch.dylib                   0x000000010daf7025 _dispatch_continuation_pop + 1022
8   libdispatch.dylib                   0x000000010db0285e _dispatch_source_invoke + 661
9   libdispatch.dylib                   0x000000010daf8efd _dispatch_queue_serial_drain + 478
10  libdispatch.dylib                   0x000000010daf9c6c _dispatch_queue_invoke + 342
11  libdispatch.dylib                   0x000000010dafcef4 _dispatch_root_queue_drain + 783
12  libdispatch.dylib                   0x000000010dafcb82 _dispatch_worker_thread4 + 54
13  libsystem_pthread.dylib             0x000000010df435a2 _pthread_wqthread + 1299
14  libsystem_pthread.dylib             0x000000010df4307d start_wqthread + 13
2017-06-09 08:55:16.594451-0700 recharge-consumer-ios[98610:1552400] [reports] Main Thread Checker: UI API called on a background thread: -[UIApplication applicationState]
PID: 98610, TID: 1552400, Thread name: (none), Queue name: com.sift.SFUploader, QoS: 0
Backtrace:
4   Sift                                0x000000010851874c -[SFUploader doUpload] + 76
5   Sift                                0x0000000108517aec __58-[SFUploader initWithArchivePath:sift:config:backoffBase:]_block_invoke + 44
6   libdispatch.dylib                   0x000000010daf06c8 _dispatch_client_callout + 8
7   libdispatch.dylib                   0x000000010daf7025 _dispatch_continuation_pop + 1022
8   libdispatch.dylib                   0x000000010db0285e _dispatch_source_invoke + 661
9   libdispatch.dylib                   0x000000010daf8efd _dispatch_queue_serial_drain + 478
10  libdispatch.dylib                   0x000000010daf9c6c _dispatch_queue_invoke + 342
11  libdispatch.dylib                   0x000000010dafcef4 _dispatch_root_queue_drain + 783
12  libdispatch.dylib                   0x000000010dafcb82 _dispatch_worker_thread4 + 54
13  libsystem_pthread.dylib             0x000000010df435a2 _pthread_wqthread + 1299
14  libsystem_pthread.dylib             0x000000010df4307d start_wqthread + 13
```

Xcode 9.0 Beta (9M136h)
Sift pod 0.9.7

A couple other projects where the same issue has been reported:
https://github.com/mixpanel/mixpanel-iphone/issues/677
https://github.com/firebase/firebase-ios-sdk/issues/69

**Comment by garylee1** [comment]
thanks @edenman – we'll put this on our radar.

**Comment by garylee1** [comment]
Got a fix for this going out soon!

**Comment by garylee1** [comment]
@edenman this has been fixed and released with v0.9.8 – see the changes here:

https://github.com/SiftScience/sift-ios/commit/1ac75a68f866acdae858d197169cd9d01bc75ec4#diff-06af5c43f3f44db25f2cf36bd0061cf5
